datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1925,12 +1925,29 @@ pub fn array_length(args: &[ArrayRef]) -> Result<ArrayRef> {
 
 /// Array_dims SQL function
 pub fn array_dims(args: &[ArrayRef]) -> Result<ArrayRef> {
-    let list_array = as_list_array(&args[0])?;
+    let data = match args[0].data_type() {
```

Comment:
please check args not empty, especially if its pub function

Reply:
I will address this later in the pull request as other functions may also lack this check.